security/security.c

```diff
@@ -696,27 +777,29 @@ static void __init lsm_early_task(struct task_struct *task)
  * call_int_hook:
  *	This is a hook that returns a value.
  */
+#define CALL_STATIC_SLOT_VOID(NUM, HOOK, ...)				\
+	static_call_cond(STATIC_SLOT(HOOK, NUM))(__VA_ARGS__);
 
-#define call_void_hook(FUNC, ...)				\
-	do {							\
-		struct security_hook_list *P;			\
-								\
-		hlist_for_each_entry(P, &security_hook_heads.FUNC, list) \
-			P->hook.FUNC(__VA_ARGS__);		\
+#define call_void_hook(FUNC, ...)					\
+	do {								\
+		SECURITY_FOREACH_STATIC_SLOT(CALL_STATIC_SLOT_VOID, 	\
+					     FUNC, __VA_ARGS__)		\
 	} while (0)
 
-#define call_int_hook(FUNC, IRC, ...) ({			\
-	int RC = IRC;						\
-	do {							\
-		struct security_hook_list *P;			\
-								\
-		hlist_for_each_entry(P, &security_hook_heads.FUNC, list) { \
-			RC = P->hook.FUNC(__VA_ARGS__);		\
-			if (RC != 0)				\
-				break;				\
-		}						\
-	} while (0);						\
-	RC;							\
+#define CALL_STATIC_SLOT_INT(NUM, R, HOOK, ...)				\
```

Comment:
This is a super unhygienic macro - I think that's unavoidable but let's put some `__` at the beginning of the name so nobody mistakes it for a "proper" macro!